javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -451,9 +451,80 @@ object Integer {
   @inline def valueOf(s: String, radix: scala.Int): Integer =
     valueOf(parseInt(s, radix))
 
-  // TODO:
-  // def parseUnsignedInt(s: String): scala.Int = parseUnsignedInt(s, 10)
-  // def parseUnsignedInt(s: String, radix: scala.Int): scala.Int = ???
-  // def toUnsignedString(i: scala.Int): String = toUnsignedString(i, 10)
-  // def toUnsignedString(_i: scala.Int, _radix: scala.Int): String = ???
+  @inline def parseUnsignedInt(s: String): scala.Int = parseUnsignedInt(s, 10)
+
+  def parseUnsignedInt(s: String, radix: scala.Int): scala.Int = {
+    if (s == null || radix < Character.MIN_RADIX ||
+        radix > Character.MAX_RADIX) throw new NumberFormatException(s)
+
+    val len = s.length()
+
+    if (len == 0) throw new NumberFormatException(s)
+
+    val hasPlusSign = s.charAt(0) == '+'
+
+    if (hasPlusSign && len == 1) throw new NumberFormatException(s)
+
+    val offset = if (hasPlusSign) 1 else 0
+
+    parseUnsigned(s, offset, radix)
+  }
+
+  private def parseUnsigned(s: String, _offset: Int, radix: Int): scala.Int = {
+    val unsignedIntMaxValue = -1
+    val max                 = divideUnsigned(unsignedIntMaxValue, radix)
+    var result = 0
+    var offset = _offset
+    val length = s.length()
+
+    while (offset < length) {
+      val digit = Character.digit(s.charAt(offset), radix)
+      offset += 1
+
+      if (digit == -1) throw new NumberFormatException(s)
+
+      if (compareUnsigned(result, max) == 1) throw new NumberFormatException(s)
+
+      result = result * radix + digit
+    }
```

Comment:
One test case for you: `Integer.parseUnsignedInt("4294967296")`. It should fail with `NumberFormatException`, but currently this will succeed and return 0. That's because the (unsigned) addition overflows.
You need to test that, after the addition of `digit`, `result` is not smaller than `digit` (which would signal an overflow during the addition):
```scala
if (compareUnsigned(result, digit) < 0) throw new NumberFormatException(s)
```